src/openms/source/METADATA/MetaInfo.cpp

```diff
@@ -78,13 +78,22 @@ namespace OpenMS
   void MetaInfo::setValue(const String & name, const DataValue & value)
   {
     UInt index = registry_.registerName(name); // no-op if name is already registered
-    index_to_value_[index] = value;
+    setValue(index, value);
   }
 
   void MetaInfo::setValue(UInt index, const DataValue & value)
   {
     // @TODO: check if that index is registered in MetaInfoRegistry?
-    index_to_value_[index] = value;
+    auto it = index_to_value_.find(index);
+    if ( it != index_to_value_.end()) 
+    {
+      it->second = value; 
+    }
+    else
+    {
+      DataValue tmp = value;
```

Comment:
@timosachsenberg I think you should add some documentation / explanation for this here, otherwise it is unclear why this was done and will be reverted at some point